src/main/java/net/neoforged/neoforge/capabilities/BlockCapability.java

```diff
@@ -120,11 +122,46 @@ public final class BlockCapability<T, C extends @Nullable Object> extends BaseCa
 
     /**
      * {@return a new immutable copy of all the currently known block capabilities}
+     *
+     * <p>Mods that want to forward "all" capability requests should likely use {@link #getAllProxyable()} instead.
      */
     public static synchronized List<BlockCapability<?, ?>> getAll() {
         return registry.getAll();
     }
 
+    /**
+     * {@return a new immutable copy of all the currently known proxyable block capabilities}
+     *
+     * <p>This method should typically only be used in the {@link EventPriority#LOW} or
+     * {@link EventPriority#LOWEST} phase of {@link RegisterCapabilitiesEvent}, or later,
+     * to ensure that all proxyable capabilities have been {@link RegisterCapabilitiesEvent#setProxyable marked as such}.
+     *
+     * @see #isProxyable()
+     */
+    public static synchronized List<BlockCapability<?, ?>> getAllProxyable() {
```

Comment:
Why is this only implemented on block capabilities? What about proxying item capabilities or entity? As an example, I have a design for a block that proxies a item fluid handler to a block entity.

Reply:
No, I am not. It proxies an item fluid handler from any source. Could be my item, could be your item. Could be a bucket.

I don't see how its different than having my block proxy cap calls to your block.

Reply:
ItemStacks are not context-aware, so they don't really have the same limitations. It's effectively the same as if you passed the ItemStack to the caller and let them pull the capability from it - the ItemStack is none the wiser.

BlockEntities, otoh, _are_ context aware and might do things on the presumption that something happened from a given direction relative to their blockpos. Entities are also context sensitive, and could benefit (though, I doubt it matters, nobody is really doing a blanket-proxy of entities like mods are for blocks), but ItemStacks are not.